__tests__/Unit/Components/Searchbar/searchbar.test.tsx

```diff
@@ -0,0 +1,64 @@
+import Searchbar from '@/components/Dashboard/Searchbar';
+import { splitNSearch } from '@/utils/splitNSearch';
+import { fireEvent, render, screen } from '@testing-library/react';
+
+jest.mock('../../../../src/utils/splitNSearch', () => ({
+    splitNSearch: jest.fn(),
+}));
+
+describe('test searchbar component', function () {
+    it('renders searchbar input and a search button', function () {
+        render(<Searchbar label="test-label" />);
+
+        const input = screen.getByTestId('searchbar_input') as HTMLInputElement;
+        const searchBtn = screen.getByTestId('search_btn') as HTMLButtonElement;
+
+        expect(input).toBeInTheDocument();
+        expect(input).toHaveAttribute('placeholder', 'test-label:');
+        expect(searchBtn).toBeInTheDocument();
+        expect(searchBtn).toHaveTextContent('Search');
+    });
+
+    it('checks if we can type into the searchbar', function () {
+        render(<Searchbar label="test-label" />);
+
+        const input = screen.getByTestId('searchbar_input') as HTMLInputElement;
+
+        fireEvent.change(input, { target: { value: '123,456' } });
+        expect(input.value).toBe('123,456');
+    });
+
+    it('tests if the click handler is called', function () {
+        render(<Searchbar label="test-label" />);
+
+        const input = screen.getByTestId('searchbar_input') as HTMLInputElement;
+
+        const searchBtn = screen.getByTestId('search_btn') as HTMLButtonElement;
+
+        fireEvent.change(input, { target: { value: '123' } });
+        fireEvent.click(searchBtn);
+
+        expect(splitNSearch).toBeCalledTimes(1);
+    });
+
+    it('tests if enter key calls search', function () {
+        render(<Searchbar label="test-label" />);
+
+        const input = screen.getByTestId('searchbar_input') as HTMLInputElement;
+
+        fireEvent.change(input, { target: { value: '123' } });
+        fireEvent.keyDown(input, { key: 'Enter', code: 'Enter', charCode: 13 });
+
+        expect(splitNSearch).toBeCalledTimes(2);
+    });
+
+    it('tests other key down events do not call search', function () {
+        render(<Searchbar label="test-label" />);
+
+        const input = screen.getByTestId('searchbar_input') as HTMLInputElement;
+
+        fireEvent.keyDown(input, { key: 'A', code: 'KeyA' });
+
+        expect(splitNSearch).toBeCalledTimes(2);
```

Comment:
Is this function `splitNSearch` called even for other keystrokes apart from Enter?

Reply:
No, but since I mocked the function before tests start and above you see that it has been once called by pressing the button and then by pressing the enter key so when the third time I check by pressing the key `A` I am ensuring the called number of times does not change than the number of times it was called before

Reply:
Then you should probably mock the function again very time before the test is executed. This is not the right way to test - it gives a sense that the function was called twice.

Reply:
> Then you should probably mock the function again very time before the test is executed. This is not the right way to test - it gives a sense that the function was called twice.

Tried that as well, its not working

Reply:
> Then you should probably mock the function again very time before the test is executed. This is not the right way to test - it gives a sense that the function was called twice.

Any doc that says so that checking if a mocked function is called cannot be picked in incrementation of previous tests.

Reply:
The main purpose of mocking functions is to test how the module being tested behaves and reacts with the function independently of external factors - i.e. for isolating test scenarios. Making the mock function dependent on the previous state of the mocked function defeats this purpose.

These are more of a practice / coding standards than facts from documentation. I have seen it being followed in production code, hence added the comment here.

Reply:
> These are more of a practice / coding standards than facts from documentation. I have seen it being followed in production code, hence added the comment here.

every industry has different rules in place, So if you could refer any strong and official source that gives this in written i will be happy to change my code

Reply:
> Like I already explained it's a generally followed coding practice.

yet again the same thing, It is not mandatory to do that

Reply:
also i would love if you could help me in achieving that as I am not able to clear my mocks

Reply:
We can connect regarding this.

Reply:
Resolving this as i have used spyon now